YJE/Day2/가운데 글자 가져오기.js

```diff
@@ -0,0 +1,11 @@
+function solution(s) {
+    let answer = '';
+    let middle = Math.floor(s.length / 2);
+    if(s.length % 2 === 0) { 
+        answer += s[middle-1] += s[middle];
```

Comment:
현재 answer += s[middle - 1] += s[middle]; 구문은 예상치 못한 결과를 가져올 수 있습니다.
예를 들어 문자열의 길이가 4일 경우, middle은 2가 되고 

`s[1] = s[1] + s[2]`와 같은 동작을 수행합니다. 이로 인해 s[1]의 원래 값이 변경되며, 이후 코드 흐름에 영향을 줄 수 있습니다.

또한 이 표현은 코드의 의도를 직관적으로 파악하기 어려워, 유지보수에도 불리합니다.

단순히 두 문자를 연결해 answer에 할당하려는 목적이라면, + 연산자를 사용하는 방식이 훨씬 명확하고 안전합니다.🙂


```suggestion
        answer = s[middle - 1] + s[middle];
```

Reply:
아 이런 문제가 발생할 수 있군요! 알려주셔서 감사합니다^^